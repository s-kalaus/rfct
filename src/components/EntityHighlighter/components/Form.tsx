import React from 'react';

export interface IProps {
  disabled: boolean;
  label: string;
  onSubmit: (label: string) => void;
  onLabelChange: (label: string) => void;
}

export const Form = ({ label, disabled, onSubmit, onLabelChange }: IProps) => (
  <div>
    <input
      data-testid="input"
      type="text"
      placeholder="Entity label"
      value={label}
      onChange={(event) => onLabelChange(event.target.value)}
      disabled={disabled}
    />
    <button
      onClick={() => onSubmit(label)}
      disabled={disabled}
      data-testid="button"
    >
      Add entity for selection
    </button>
  </div>
);
