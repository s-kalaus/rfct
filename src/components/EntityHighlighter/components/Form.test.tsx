import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Form } from './Form';

describe('Form', () => {
  const label = 'Label';
  const onSubmit = jest.fn();
  const onLabelChange = jest.fn();

  test('renders component', () => {
    const { container } = render(
      <Form
        label={label}
        disabled={false}
        onSubmit={onSubmit}
        onLabelChange={onLabelChange}
      />
    );
    expect(container).toMatchSnapshot();
  });

  test('react on label change', () => {
    render(
      <Form
        label={label}
        disabled={false}
        onSubmit={onSubmit}
        onLabelChange={onLabelChange}
      />
    );
    userEvent.type(screen.getByTestId('input'), '!');
    expect(onLabelChange).toHaveBeenCalledWith('Label!');
  });

  test('react on button click', () => {
    render(
      <Form
        label={label}
        disabled={false}
        onSubmit={onSubmit}
        onLabelChange={onLabelChange}
      />
    );
    userEvent.click(screen.getByTestId('button'));
    expect(onSubmit).toHaveBeenCalledWith('Label');
  });

  test('not react on label change if disabled', () => {
    render(
      <Form
        label={label}
        disabled={true}
        onSubmit={onSubmit}
        onLabelChange={onLabelChange}
      />
    );
    userEvent.type(screen.getByTestId('input'), '!');
    expect(onLabelChange).not.toHaveBeenCalled();
  });
});
