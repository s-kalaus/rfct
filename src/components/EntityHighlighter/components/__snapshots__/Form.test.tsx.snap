// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders component 1`] = `
<div>
  <div>
    <input
      data-testid="input"
      placeholder="Entity label"
      type="text"
      value="Label"
    />
    <button
      data-testid="button"
    >
      Add entity for selection
    </button>
  </div>
</div>
`;
